appcenter-push/android/src/main/java/com/microsoft/appcenter/reactnative/push/AppCenterReactNativePushModule.java

```diff
@@ -37,11 +58,20 @@ public String getName() {
     }
 
     @ReactMethod
-    public void setEnabled(boolean enabled, final Promise promise) {
+    public void setEnabled(final boolean enabled, final Promise promise) {
+        if (!mPushStarted && enabled) {
+            AppCenter.start(Push.class);
+            mPushStarted = true;
+        }
         Push.setEnabled(enabled).thenAccept(new AppCenterConsumer<Void>() {
 
             @Override
             public void accept(Void result) {
+                if (enabled) {
+
+                    /* TODO expose a way to post command in background looper in native SDK to avoid accessing storage directly here. */
```

Comment:
Does this have the same condition ("storage might not be ready yet") as above?  If so, let's include that in the comment.

Reply:
@jwallra no as storage is initialized in the constructor of this class.
@achocron it should not be a blocker for this work (we already read a file in shared directly = disk access so would require more changes to the RN SDK overall to comply with strict mode) item but yes we can add a work item for that.